tutorials/stats-sensor-space/76_new_cluster_test_api.py

```diff
@@ -0,0 +1,491 @@
+    """
+    import random
+
+    _, evoked_data_a, evoked_data_b = prep_sample_data()
+
+    # Ensure evoked_data_a and evoked_data_b are of the same length
+    assert len(evoked_data_a) == len(
+        evoked_data_b
+    ), "evoked_data_a and evoked_data_b must have the same length"
+
+    # Create a list of participant indices
+    participant_indices = list(range(len(evoked_data_a)))
+
+    # Shuffle the list of participant indices
+    random.shuffle(participant_indices)
+
+    # Reorder evoked data according to the shuffled participant indices
+    shuffled_evoked_data_a = [evoked_data_a[i] for i in participant_indices]
+    shuffled_evoked_data_b = [evoked_data_b[i] for i in participant_indices]
+
+    # Combine the shuffled evoked data into a single list
+    shuffled_evoked_data = shuffled_evoked_data_a + shuffled_evoked_data_b
+
+    # Combine the original evoked data into a single list
+    original_evoked_data = evoked_data_a + evoked_data_b
+
+    return original_evoked_data, shuffled_evoked_data
+
+
+# shuffle order of pairs
+original_evoked_data, shuffled_evoked_data = create_random_paired_evokeds_list()
+# shouldn't change the results (p-value is different though?)
+
+shuffled_participant_ids, shuffled_conditions, shuffled_evoked_data = (
+    create_random_evokeds_id_condition_list()
+)
+
+
+def prepare_dataframe_for_cluster_function(
+    evokeds: list = None,
+    condition: list = None,
+    subject_index: list = None,
+):
+    """
+    Prepare a dataframe for the cluster test function.
+
+    Parameters
+    ----------
+    contrast : bool, optional
+        If True, a contrast is calculated. Default is False.
+    evokeds : list, optional
+        List of evoked objects. Default is None.
+    condition : list, optional
+        List of conditions for each evoked object. Default is None.
+    subject_index : list, optional
+        List of subject IDs. Default is None.
+
+    Returns
+    -------
+    df : DataFrame
+        The prepared DataFrame for the cluster test function.
+    """
+    # Initialize the DataFrame with evoked data
+    df = pd.DataFrame(
+        {
+            "evoked": evokeds,
+            "condition": condition if condition is not None else np.nan,
+            "subject_index": subject_index if subject_index is not None else np.nan,
+        }
+    )
+
+    return df
+
+
+# run with original data
+df = prepare_dataframe_for_cluster_function(
+    evokeds=original_evoked_data, condition=None, subject_index=None
+)
+
+df = prepare_dataframe_for_cluster_function(
+    evokeds=shuffled_evoked_data, condition=None, subject_index=None
+)
+
+df = prepare_dataframe_for_cluster_function(
+    evokeds=shuffled_evoked_data,
+    condition=shuffled_conditions,
+    subject_index=shuffled_participant_ids,
+)
+
+
+cluster_test(df)
+
+
+def cluster_test(
+    df: pd.DataFrame,
+    contrast: bool = True,
+    n_permutations: int = 10000,
+    seed: int = 1234,
```

Comment:
see how we do seed elsewhere in MNE-Python. (accepts None, int, or instance of np.random.Generator I think)

Reply:
Good point